src/AutocompleteBadges.js

```diff
@@ -0,0 +1,145 @@
+'use strict';
+
+import templates from './AutocompleteBadges.soy.js';
+import Autocomplete from 'metal-autocomplete';
+import Component from 'metal-component';
+import Soy from 'metal-soy';
+
+class AutocompleteBadges extends Component {
+
+	/**
+	 * @inheritDoc
+	 */
+	attached() {
+		this.inputElement = this.element.querySelector('.autocomplete-field');
```

Comment:
I mentioned in another comment that this could just be a regular property instead of a state property. I've noticed now that it doesn't even need to be a property though, and we can also get rid of these calls to `querySelector`, which are slow.

You can do this by using a new feature that was recently added to metal.js, which is the ability to add `ref` to elements in the soy template, which will cause Metal.js to save a reference to that element for you to access in your js. For example, in **.autocomplete-field**'s case you could do something like this in your soy:

```
{template .render}
  <div class="autocomplete-badges">
    <textarea ref="input" ...></textarea>
    ...
  </div>
{/template}
```

Then, when creating the `Autocomplete` instance, or on any other place where you need the textarea element, you can just do something like this:

```js
this.autocomplete_ = new Autocomplete({
  inputElement: this.refs.input,
  data: this.getFilteredElements_.bind(this)
});
```

Super useful, right? You can also do this for all the other places where you're using `querySelector`, just make sure to use a different `ref` name for each element you want to access, and then just access it within `this.refs` by the chosen name. Let me know if you have any questions about this :)